keywords/MobileRestClient.py

```diff
@@ -608,11 +603,20 @@ def add_doc(self, url, db, doc, auth=None):
         doc["updates"] = 0
 
         if auth_type == AuthType.session:
-            resp = self._session.post("{}/{}/".format(url, db), data=json.dumps(doc), cookies=dict(SyncGatewaySession=auth[1]))
+            if use_post:
+                resp = self._session.post("{}/{}/".format(url, db), data=json.dumps(doc), cookies=dict(SyncGatewaySession=auth[1]))
+            else:
+                resp = self._session.put("{}/{}/{}".format(url, db, doc["_id"]), data=json.dumps(doc), cookies=dict(SyncGatewaySession=auth[1]))
         elif auth_type == AuthType.http_basic:
-            resp = self._session.post("{}/{}/".format(url, db), data=json.dumps(doc), auth=auth)
+            if use_post:
+                resp = self._session.post("{}/{}/".format(url, db), data=json.dumps(doc), auth=auth)
```

Comment:
There all lots of places where this approach could use this approach in this file. Do you want this to go in or should I file a ticket to do a full cleanup?